api-reference/remote-run.mdx

```diff
@@ -0,0 +1,201 @@
+
+class MyLLModel:
+    @cls.load
+    def load(self):
+        # load the model
+        pass
+
+    @cls.infer
+    def infer(self):
+        # run inference
+        return "generated output"
+```
+
+## Examples
+
+This section contains working examples of how to use the `remote-run` command.
+
+### Example 1 - Run a function on a remote server
+
+```python 
+# app.py
+import inferless
+
+
+@inferless.method
+def test_gpt_neo(prompt):
+    from transformers import pipeline
+    pipe = pipeline("text-generation", model="EleutherAI/gpt-neo-125m")
+    expected_output = pipe(prompt, max_length=50, do_sample=False)[0]['generated_text']
+
+    return expected_output
+
+
+def post_process(output):
+    import re
+    processed_output = re.sub(r'\b(\d{3}-\d{2}-\d{4})\b', '[REDACTED]', output)
+    return processed_output
+
+prompt = "Hello, Write a story about a dragon"
+output = test_gpt_neo(prompt)
+processed_output = post_process(output)
+print(processed_output)
+```
+    
+```yaml
+# config.yaml
+gpu:
+  T4
+build:
+    python_packages:
+        - transformers
+        - torch
+        - accelerate
+```
+run command `inferless remote-run app.py -c config.yaml`
+
+In this example, only the `test_gpt_neo` function will run on the remote server. The `post_process` function will run on the local machine.
+
+As you can see, the `transformers` is imported inside the function which means it is not required to install the transformers package on the local machine.
+
+
+<Note> For the first call, the command may take some time as it will setup the required packages on the remote server. </Note>
+
+
+### Example 2 - Run a class on a remote
+    
+```python
+# app.py
+
+import os
+import inferless
+
+InferlessCls = inferless.Cls()
+HF_TOKEN = "<HF_TOKEN>"
+
+class InferlessPythonModel:
+
+    @InferlessCls.load
+    def initialize(self):
+        from vllm import LLM, SamplingParams
+        from transformers import AutoTokenizer
+
+        model_id = "TheBloke/Llama-2-7B-chat-AWQ"
+        tokenizer_model = "meta-llama/Llama-2-7b-chat-hf"
+        self.sampling_params = SamplingParams(
+            temperature=0.7,
+            top_p=0.1,
+            repetition_penalty=1.18,
+            top_k=40,
+            max_tokens=512,
+        )
+        self.llm = LLM(model=model_id, enforce_eager=True, quantization="AWQ")
+        self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_model, token=HF_TOKEN)
+
+    @InferlessCls.infer
+    def infer(self, inputs):
+        prompts = inputs["prompt"]
+        input_text = self.tokenizer.apply_chat_template([{"role": "user", "content": prompts}], tokenize=False)
+        result = self.llm.generate(input_text, self.sampling_params)
+        result_output = [output.outputs[0].text for output in result]
+        return {"generated_result": result_output[0]}
+
+
+model = InferlessPythonModel()
+print(model.infer({"prompt": "Hello, Write a story about a dragon"}))
+```
+
+```yaml
+# config.yaml
+gpu:
```

Comment:
can we take in this in the class for function param, as I don't want to have a file specified in the existing config.yaml

Reply:
moved gpu param to the decorator